### ISSUE TITLE
Noise reduction

### ISSUE THREAD

**Issue by robintown** [vector-im/element-call#714]
#### What would you like to do?

Automatically apply noise reduction filters to participants' audio.

#### Why would you like to do it?

While noise reduction doesn't improve the intelligibility of speech, it can help prevent auditory fatigue.

#### How would you like to achieve it?

Integrate noise reduction software such as [RNNoise](https://jmvalin.ca/demo/rnnoise/) or [DeepFilterNet](https://github.com/Rikorose/DeepFilterNet) into Element Call via WASM. There should also be an option to turn it off.

### Have you considered any alternatives?

Relying on the browser's built-in noise reduction, which is usually very subtle if it exists at all.

**Comment by robintown** [comment]
After some investigation, DeepFilterNet is probably not viable for us to use due to its size. The neuron weights are already 8 MB large, and the full binary weighs in at about 20 MB. RNNoise on the other hand is only 200 kB or so at most.

We could potentially offer DeepFilterNet as a higher quality option built into the desktop and mobile clients, just not on web. RNNoise is already pretty decent, so I wouldn't spend any effort on that for now though.

cc @fkwp

**Comment by bblacher** [comment]
This would be amazing!

**Comment by itsCodeAlchemist** [comment]
+1 for this! Recently moved to Element :)
This is the only thing preventing my team to fully migrate to Element at the moment.

We don't note any difference with the current Noise Suppression and Echo Cancellation features.

**Comment by Mustache84** [comment]
Need noise suppression badly. It's so awful right now. Everybody who tried to use Element quickly dumped it b/c as a bunch of gamers, nobody wants to hear keyboards and mice or people picking up drinks and snacks.

**Comment by RoSsIsChAmP** [comment]
I want to bump this again, all my users complain of this, especially those that are coming from Discord.

**Comment by bajawoojie** [comment]
Support!! I'm currently trying to get friends to migrate from discord but between this and lack of push to talk it's rather hard to convince them.

**Comment by osousa** [comment]
On my desktop I use DeepFilterNet  with LADSPA plugin with pipewire filter-chain integration, and the results are incredible. 

Having this builtin would be a game changer for those who can't or don't have the patience to set it up at the system level.

**Comment by wendellmeset** [comment]
This would be amazing

**Comment by NepuNeptuneNep** [comment]
Yeah, you can hear every keyboard press and breathing in calls. It would help a lot.

**Comment by xaetacorenet** [comment]
It would be super nice because we have a ton of people who came from discord and this is the biggest problem they have, Adding something like Speex or Rnoise would be really nice

**Comment by xaetacorenet** [comment]
Livekit has support for this, element call uses it
https://docs.livekit.io/transport/media/noise-cancellation/
All we need is someone who know how to implement this in the livekit client code of element call since the livekit server accepts this

**Comment by xaetacorenet** [comment]
> Livekit has support for this, element call uses it https://docs.livekit.io/transport/media/noise-cancellation/ All we need is someone who know how to implement this in the livekit client code of element call since the livekit server accepts this

Seems like most of these solutions are paid, Might be some third party plugins worth investigating that could achieve the same but krisp and ai-coustics could be an issue

**Comment by xaetacorenet** [comment]
RNNoise and DeepFilterNet could be used maybe by using Livekit's AudioFrame API

**Comment by Henry-sketch** [comment]
I also want to bump this request. I am currently also migrating a group of discord users to matrix and we successfully tested the calls feature. But the missing background noise reduction on the desktop app is probably a deal breaker. 
I've tested the sable client and the recently implemented RNNoise suppression seems to have improved the experience already.

**Comment by xaetacorenet** [comment]
> I also want to bump this request. I am currently also migrating a group of discord users to matrix and we successfully tested the calls feature. But the missing background noise reduction on the desktop app is probably a deal breaker. I've tested the sable client and the recently implemented RNNoise suppression seems to have improved the experience already.

We have a fork that has this https://github.com/NeonDystopiaWorld/neo-element-web/releases/tag/neo-v0.1.0
